mlrun/launcher/base.py

```diff
@@ -313,3 +340,20 @@ def _are_valid_notifications(runobj) -> bool:
             return False
 
         return True
+
+    def _refresh_function_metadata(self, runtime: "mlrun.runtimes.BaseRuntime"):
+        try:
+            meta = runtime.metadata
+            db_func = self.db.get_function(meta.name, meta.project, meta.tag)
+            if db_func and "status" in db_func:
+                runtime.status = db_func["status"]
+                if (
+                    runtime.status.state
+                    and runtime.status.state == "ready"
+                    and not hasattr(runtime.status, "nuclio_name")
+                ):
```

Comment:
is that the way to say ?
1. runtime is ready
2. not nuclio

if so, can think of a better readable way doing it? if not, leave as is but perhaps comment that

Reply:
I'm not sure, if it is then we can use `runtime.kind in RuntimeKinds.nuclio_runtimes()`
but why wouldn't we update the image if it's nuclio runtime? maybe @Tankilevitch will have an idea

Reply:
added a comment in the code explaining why we use this attribute